App.tsx

```diff
@@ -5,17 +5,36 @@
  * @format
  */
 
-import * as React from 'react';
-import {NavigationContainer} from '@react-navigation/native';
+import React from 'react';
+import {
+  NavigationContainer,
+  NavigatorScreenParams,
+} from '@react-navigation/native';
 import {createBottomTabNavigator} from '@react-navigation/bottom-tabs';
 import HomeScreen from './screens/HomeScreen';
 import ProductsScreen from './screens/ProductsScreen';
 import CartScreen from './screens/CartScreen';
 import ProductDetails from './screens/ProductDetails';
 import {createNativeStackNavigator} from '@react-navigation/native-stack';
 
-const Tab = createBottomTabNavigator();
-const Stack = createNativeStackNavigator();
+export type TabNavigationParamList = {
+  Home: undefined;
+  Products: undefined;
+  Cart: undefined;
+};
+
+export type RootStackParamList = {
+  BottomTabs: NavigatorScreenParams<TabNavigationParamList>;
+  ProductDetails: {
+    productTitle: string;
+    productDescription: string;
+    productPrice: number;
+    productImage: string;
+  };
+};
+
+const Tab = createBottomTabNavigator<TabNavigationParamList>();
+const Stack = createNativeStackNavigator<RootStackParamList>();
```

Comment:
See https://reactnavigation.org/docs/typescript/ on how to use React Navigation with Typescript.